### PR TITLE
Apply clang-tidy

### PR DESCRIPTION
**Pull request by Schneegans**
It still reports many additional warnings, but these are the ones which could be applied automatically and semi-automatically.

### CODE REVIEW

File: plugins/csp-measurement-tools/src/voronoi/Breakpoint.hpp [hunk 1]
```diff
@@ -24,15 +24,15 @@ class Breakpoint {
   Vector2f const& position() const;
   Edge            finishEdge(Vector2f const& end) const;
 
-  Arc *       mLeftArc, *mRightArc;
-  Breakpoint *mLeftChild, *mRightChild, *mParent;
+  Arc *       mLeftArc{nullptr}, *mRightArc{nullptr};
+  Breakpoint *mLeftChild{nullptr}, *mRightChild{nullptr}, *mParent{nullptr};
```
Comment (JonasGilg) [inline plugins/csp-measurement-tools/src/voronoi/Breakpoint.hpp:28]: This is a rule we enabled, but clang-tidy can't apply it to member variables due to technical limitations. Especially with the use of pointer we should enforce that.
```suggestion
  Arc* mLeftArc{nullptr};
  Arc* mRightArc{nullptr};
  
  Breakpoint* mLeftChild{nullptr};
  Breakpoint* mRightChild{nullptr};
  Breakpoint* mParent{nullptr};
```

File: plugins/csp-measurement-tools/src/voronoi/VoronoiGenerator.hpp [hunk 2]
```diff
@@ -49,8 +49,8 @@ class VoronoiGenerator {
   void finishEdges();
 
   Beachline mBeachline;
-  double    mSweepline;
-  double    mMaxY, mMinY;
+  double    mSweepline{0.0};
+  double    mMaxY{0.0}, mMinY{0.0};
```
Comment (JonasGilg) [inline plugins/csp-measurement-tools/src/voronoi/VoronoiGenerator.hpp:53]:
```suggestion
  double    mMaxY{0.0};
  double    mMinY{0.0};
```

File: src/cs-graphics/internal/tiny_gltf_helper.hpp [hunk 3]
```diff
@@ -18,61 +18,74 @@ namespace cs::graphics::internal {
...
-  } else if (accessor.type == TINYGLTF_TYPE_MAT3) {
+  }
+  if (accessor.type == TINYGLTF_TYPE_MAT3) {
     return 9;
-    //} else if (accessor.type == tinygltf::TINYGLTF_TYPE_MAT4) { return 16;
-  } else {
-    return 16;
   }
+  // if (accessor.type == tinygltf::TINYGLTF_TYPE_MAT4) { return 16;
+  return 16;
```
Comment (JonasGilg) [inline src/cs-graphics/internal/tiny_gltf_helper.hpp:38]:
```suggestion
  }
  // accessor.type == tinygltf::TINYGLTF_TYPE_MAT4
  return 16;
```

**Review by JonasGilg — CHANGES_REQUESTED** [review]

**Comment by JonasGilg** [comment]
> It still reports many additional warnings, but these are the ones which could be applied automatically and semi-automatically.

But no errors reported, right? Because all relevant warnings are errors. Most of the warnings are warnings, because there are instances which couldn't be fixed due to third party libraries.

**Comment by Schneegans** [comment]
I fixed many but not all warnings. I think if it builds fine in CI, it's already a good step forward nonetheless.

File: plugins/csp-trajectories/src/Plugin.hpp [hunk 4]
```diff
@@ -34,7 +34,8 @@ class Plugin : public cs::core::PluginBase {
 
         /// Without this, some versions of clang claim this type to be non-default-constructible...
         /// Also, Trail() = default; does not work...
-        Trail(){};
+        Trail() = default;
+        ;
```
Comment (JonasGilg) [inline plugins/csp-trajectories/src/Plugin.hpp]: Either the comment or the code is wrong now.

**Review by JonasGilg — CHANGES_REQUESTED** [review]

File: plugins/csp-trajectories/src/Plugin.hpp (hunk 4 above)
Comment (Schneegans) [reply plugins/csp-trajectories/src/Plugin.hpp]: This was "auto-fixed" by clang-tidy. Now it should work again... MSVC still complains. I'll see about that on Monday.

**Review by JonasGilg — APPROVED** [review]

**Pull request merged**